packages/hooks/src/useOnKeyDown/useOnKeyDown.ts

```diff
@@ -0,0 +1,53 @@
+import useEventListener from "@use-it/event-listener";
+import { XOR } from "ts-xor";
+
+type SimpleKeyComparerator = globalThis.KeyboardEvent["key"];
+
+interface VerboseKeyComparerator {
+  readonly key: SimpleKeyComparerator;
+  readonly shiftKey?: boolean;
+  readonly ctrlKey?: boolean;
+  readonly altKey?: boolean;
+  readonly metaKey?: boolean;
+  readonly [index: string]: boolean | string | undefined;
+}
+
+type KeyComparerator = XOR<VerboseKeyComparerator, SimpleKeyComparerator>;
+
+export function useOnKeyDown(
+  callback: (event: globalThis.KeyboardEvent) => void,
+  keys: KeyComparerator[] | KeyComparerator,
+) {
+  const handler: (event: globalThis.KeyboardEvent) => void = (
```

Comment:
Our conventions would prefer this function defined with the `function` keyword.

```suggestion
function handler(event: globalThis.KeyboardEvent) {
```